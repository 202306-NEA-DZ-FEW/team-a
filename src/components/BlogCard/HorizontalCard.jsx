import Image from "next/image";
import Link from "next/link";
import { useTranslation } from "next-i18next";
import { IoIosArrowDropright } from "react-icons/io";

import { formatDate, truncateString } from "@/lib/helpers";

function HorizontalCard({ title, description, imageUrl, createdAt, id }) {
  const { t } = useTranslation();
  const date = formatDate(createdAt, "MMM YYYY");
  const truncatedDescription = truncateString(description, 200);
  return (
    <div className='card lg:card-side bg-base-100 shadow-xl h-full lg:h-[26rem]'>
      <figure className=' lg:w-[45%]'>
        <Image
          height={400}
          width={400}
          priority
          src={imageUrl}
          alt='Blog Image'
          className='w-full h-full'
        />
      </figure>
      <div className='card-body lg:w-[35%] p-6 lg:p-10 gap-4'>
        <h2 className='card-title text-2xl lg:text-3xl' title={title}>
          {title}
        </h2>
        <p
          className=' text-md lg:text-xl tracking-wide font-light'
          title={description}
        >
          {truncatedDescription}
        </p>
        <div className='card-actions flex gap-2 justify-between'>
          <span>{date}</span>
          <Link
            href={`/blogs/${id}`}
            className='text-secondary font-bold flex items-center gap-2'
          >
            {t("common:buttons:readMore")}

            <IoIosArrowDropright className='text-xl' />
          </Link>
        </div>
      </div>
    </div>
  );
}
export default HorizontalCard;
