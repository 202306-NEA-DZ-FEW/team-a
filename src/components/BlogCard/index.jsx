import Image from "next/image";
import Link from "next/link";
import { useTranslation } from "next-i18next";
import { IoIosArrowDropright } from "react-icons/io";

import { formatDate, truncateString } from "@/lib/helpers";

function BlogCard({ id, title, imageUrl, createdAt, description }) {
  const { t } = useTranslation();
  const date = formatDate(createdAt, "MMM YYYY");
  const truncatedDescription = truncateString(description, 95);
  const truncatedTitle = truncateString(title, 23);
  return (
    <div className='card rounded-xl shadow-lg w-full lg:w-[50%]'>
      <figure className='relative h-[60%]'>
        <Image
          className='w-full object-cover rounded-t-xl'
          src={imageUrl}
          alt='Blog Image'
          height={400}
          width={400}
          priority
        />
      </figure>

      <div className='gap-2 flex flex-col p-6'>
        <h2 className='text-3xl lg:text-4xl font-bold mb-2'>{title}</h2>
        <p className='text-md lg:text-lg font-light mb-4' title={description}>
          {truncatedDescription}
        </p>
        <div className='flex justify-between pb-4'>
          <span>{date} </span>
          <Link
            href={`/blogs/${id}`}
            className='text-secondary hover:text-secondary font-bold flex items-center gap-2'
          >
            {t("common:buttons:readMore")}
            <IoIosArrowDropright className='text-xl' />
          </Link>
        </div>
      </div>
    </div>
  );
}

export default BlogCard;
