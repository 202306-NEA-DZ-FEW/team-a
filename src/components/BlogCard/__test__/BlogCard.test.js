import renderer from "react-test-renderer";

import BlogCard from "..";

jest.mock("@/lib/helpers", () => ({
  ...jest.requireActual("@/lib/helpers"),
  truncateString: jest.fn().mockReturnValue("Mocked Truncate String"),
  formatDate: jest.fn().mockReturnValue("2023-10-28", "MMM YYYY"),
}));

it("renders correctly", () => {
  const tree = renderer
    .create(
      <BlogCard
        createdAt='2023-10-28'
        id={1}
        imageUrl='https://placehold.co/600x400'
        title='title'
        description='description'
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
