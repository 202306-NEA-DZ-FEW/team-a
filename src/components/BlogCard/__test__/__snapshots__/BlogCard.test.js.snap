// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="card rounded-xl shadow-lg w-full lg:w-[50%]"
>
  <figure
    className="relative h-[60%]"
  >
    <img
      alt="Blog Image"
      className="w-full object-cover rounded-t-xl"
      data-nimg="1"
      decoding="async"
      fetchpriority="high"
      height={400}
      onError={[Function]}
      onLoad={[Function]}
      src="/_next/image?url=https%3A%2F%2Fplacehold.co%2F600x400&w=828&q=75"
      srcSet="/_next/image?url=https%3A%2F%2Fplacehold.co%2F600x400&w=640&q=75 1x, /_next/image?url=https%3A%2F%2Fplacehold.co%2F600x400&w=828&q=75 2x"
      style={
        Object {
          "color": "transparent",
        }
      }
      width={400}
    />
  </figure>
  <div
    className="gap-2 flex flex-col p-6"
  >
    <h2
      className="text-3xl lg:text-4xl font-bold mb-2"
    >
      title
    </h2>
    <p
      className="text-md lg:text-lg font-light mb-4"
      title="description"
    >
      Mocked Truncate String
    </p>
    <div
      className="flex justify-between pb-4"
    >
      <span>
        2023-10-28
         
      </span>
      <a
        className="text-secondary hover:text-secondary font-bold flex items-center gap-2"
        href="/blogs/1"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        buttons.readMore
        <svg
          className="text-xl"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M210.7 147.6c7.5-7.5 19.8-7.5 27.3 0l95.4 95.7c7.3 7.3 7.5 19.1.6 26.6l-94 94.3c-3.8 3.8-8.7 5.7-13.7 5.7-4.9 0-9.9-1.9-13.6-5.6-7.5-7.5-7.6-19.7 0-27.3l79.9-81.1-81.9-81.1c-7.6-7.4-7.6-19.6 0-27.2z"
          />
          <path
            d="M48 256c0 114.9 93.1 208 208 208s208-93.1 208-208S370.9 48 256 48 48 141.1 48 256zm32 0c0-47 18.3-91.2 51.6-124.4C164.8 98.3 209 80 256 80s91.2 18.3 124.4 51.6C413.7 164.8 432 209 432 256s-18.3 91.2-51.6 124.4C347.2 413.7 303 432 256 432s-91.2-18.3-124.4-51.6C98.3 347.2 80 303 80 256z"
          />
        </svg>
      </a>
    </div>
  </div>
</div>
`;
